crates/ecs/src/lib.rs

```diff
@@ -1300,246 +1324,304 @@ mod tests {
         world.add_component_to_entity(entity3, 3_u32).unwrap();
         world.add_component_to_entity(entity3, 3_i32).unwrap();
 
-        let k: &Archetype = world.archetypes.get(2).unwrap();
+        (world, entity1, entity2, entity3)
+    }
 
-        let x = k
+    fn get_u32_and_i32_component_vectors(
+        archetype: &Archetype,
+    ) -> (ReadComponentVec<u32>, ReadComponentVec<i32>) {
+        let component_vec_u32 = archetype
             .component_typeid_to_component_vec
             .get(&TypeId::of::<u32>())
             .unwrap()
-            .deref();
-        let comp_vec_1 = borrow_component_vec::<u32>(x).unwrap();
+            .as_ref();
+        let u32_read_vec = borrow_component_vec::<u32>(component_vec_u32);
 
-        let y = k
+        let component_vec_i32 = archetype
             .component_typeid_to_component_vec
             .get(&TypeId::of::<i32>())
             .unwrap()
-            .deref();
-        let comp_vec_2 = borrow_component_vec::<i32>(y).unwrap();
-
-        let value_1_1 = comp_vec_1.get(0).unwrap().unwrap();
-        let value_2_1 = comp_vec_1.get(1).unwrap().unwrap();
-        let value_3_1 = comp_vec_1.get(2).unwrap().unwrap();
-        let value_1_2 = comp_vec_2.get(0).unwrap().unwrap();
-        let value_2_2 = comp_vec_2.get(1).unwrap().unwrap();
-        let value_3_2 = comp_vec_2.get(2).unwrap().unwrap();
-
-        // Check original values and indexes
-
-        assert_eq!(value_1_1, 1);
-        assert_eq!(value_2_1, 2);
-        assert_eq!(value_3_1, 3);
-        assert_eq!(value_1_2, 1);
-        assert_eq!(value_2_2, 2);
-        assert_eq!(value_3_2, 3);
-
-        assert_eq!(k.entity_order, vec![entity1, entity2, entity3]);
-        assert_eq!(
-            k.entity_to_component_index.get(&entity1).unwrap().clone(),
-            0
-        );
-        assert_eq!(
-            k.entity_to_component_index.get(&entity2).unwrap().clone(),
-            1
-        );
-        assert_eq!(
-            k.entity_to_component_index.get(&entity3).unwrap().clone(),
-            2
-        );
-
-        drop(comp_vec_1);
-        drop(comp_vec_2);
+            .as_ref();
+        let i32_read_vec = borrow_component_vec::<i32>(component_vec_i32);
+
+        (u32_read_vec, i32_read_vec)
+    }
+
+    #[test]
+    fn entities_contain_correct_values_after_adding_components() {
+        let (world, _entity1, _entity2, _entity3) =
+            setup_world_with_3_entities_with_u32_and_i32_components();
+
+        let archetype = world.archetypes.get(2).unwrap();
+
+        let (u32_read_vec, i32_read_vec) = get_u32_and_i32_component_vectors(archetype);
+        let u32_values = u32_read_vec.unwrap();
+        let i32_values = i32_read_vec.unwrap();
```

Comment:
nit:

since you always unwrap the returned values, you should unwrap them within the function `get_u32_and_i32_component_vectors` and return the unwrapped `(u32_values, i32_values)`